### PR TITLE
Gothi spell changes

### PR DESCRIPTION
**Pull request by jpawlicki**
Spells have a 66% chance to continue after the vote ceases.
Spells have a grace period in which they don’t damage crops, but the damage increases the longer the spell runs.

### CODE REVIEW

**Comment by jpawlicki** [comment]
Also, the warwinds attrition is only done once.
Also, the deluge has to be active for a turn before navies can move between land regions.
(This is just to clean up the need to have land↔land order options during the non-deluge...)

File: frontend/orders-pane.js [hunk 1]
```diff
@@ -48,10 +48,10 @@ class OrdersPane extends HTMLElement {
...
+					<label id="gothi_Alyrja"><input type="checkbox" name="gothi_Alyrja"/>Vote to summon the <tooltip-element tooltip="The warwinds stops sea trade, destroys 25% of any army or navy at sea, and blows vessels in sea regions to random adjacent regions. It will start to destroy crops worldwide after 2 weeks of activity.">Warwinds</tooltip-element></label>
+					<label id="gothi_Rjinku"><input type="checkbox" name="gothi_Rjinku"/>Vote to summon the <tooltip-element tooltip="Each construction has a 33% chance of being destroyed each week the quake is active. It will start to destroy crops worldwide after 2 weeks of activity.">Quake</tooltip-element></label>
+					<label id="gothi_Syrjen"><input type="checkbox" name="gothi_Syrjen"/>Vote to summon the <tooltip-element tooltip="The deluge makes all land regions treacherous, allows navies to traverse land regions and participate in battles there (and prevents them from being captured). It will start to destroy crops worldwide after 2 weeks of activity.">Deluge</tooltip-element></label>
```
Comment (pawlick2) [inline frontend/orders-pane.js]: Wasn't climate removed in a previous PR?  Should this include the effect of making land regions treacherous?

File: backend/src/main/java/com/empire/World.java [hunk 2]
```diff
@@ -761,33 +761,50 @@ void setGothiSpellStatus() {
...
+					tivar.quake++;
+				}
 			}
 			if (passedSpells.contains("Syrjen")) {
-				if (!tivar.deluge) notifyAllPlayers("The Deluge Begins", "The necessary number of gothi of Syrjen have agreed to call forth the Deluge. Terrible floods make all land regions treacherous (if navigable by fleets), and more crops are drowned in the muddy waters every week.");
-				tivar.deluge = true;
-			} else if (tivar.deluge) {
-				tivar.deluge = Math.random() < 0.5;
-				if (!tivar.deluge) notifyAllPlayers("Deluge Ends", "The magical deluge drowning the land has finally ceased.");
+				if (tivar.deluge == 0) notifyAllPlayers("The Deluge Begins", "The necessary number of gothi of Syrjen have agreed to call forth the Deluge. Terrible floods make all land regions treacherous (if navigable by fleets), and more crops are drowned in the muddy waters every week.");
```
Comment (pawlick2) [inline backend/src/main/java/com/empire/World.java]: Another reference here to treacherous land

File: backend/src/main/java/com/empire/World.java [hunk 3]
```diff
@@ -761,33 +761,50 @@ void setGothiSpellStatus() {
...
+					tivar.deluge++;
+				}
 			}
 			if (passedSpells.contains("Lyskr")) {
-				if (!tivar.veil) notifyAllPlayers("The Veil Falls", "The necessary number of gothi of Lyskr have agreed to call forth the Veil. Heavy fog chokes out the sun and reduces visibility severely. Barred from the sunlight, crops will be lost every week until the fog lifts.");
-				tivar.veil = true;
-			} else if (tivar.veil) {
-				tivar.veil = Math.random() < 0.5;
-				if (!tivar.veil) notifyAllPlayers("Veil Ends", "The magical fog blanketing the land has finally lifted.");
+				if (tivar.veil == 0) notifyAllPlayers("The Veil Falls", "The necessary number of gothi of Lyskr have agreed to call forth the Veil. Heavy fog chokes out the sun and reduces visibility severely. Barred from the sunlight, crops will be lost every week until the fog lifts.");
```
Comment (pawlick2) [inline backend/src/main/java/com/empire/World.java]: This description seems to contradict the idea of the crop-destroying grace period.

File: backend/src/main/java/com/empire/World.java [hunk 4]
```diff
@@ -761,33 +761,50 @@ void setGothiSpellStatus() {
...
+					tivar.veil++;
+				}
 			}
 			if (passedSpells.contains("Alyrja")) {
-				if (!tivar.warwinds) notifyAllPlayers("The Warwinds Howl", "The necessary number of gothi of Alyrja have agreed to call forth the Warwinds. Titanic waves make seagoing treacherous, and powerful winds blow all vessels at sea off course. The temperature plummets, killing crops every week until the storm ceases.");
-				tivar.warwinds = true;
-			} else if (tivar.warwinds) {
-				tivar.warwinds = Math.random() < 0.5;
-				if (!tivar.warwinds) notifyAllPlayers("Warwinds End", "The magical storms devastating the land have finally calmed.");
+				if (tivar.warwinds == 0) notifyAllPlayers("The Warwinds Howl", "The necessary number of gothi of Alyrja have agreed to call forth the Warwinds. Titanic waves make seagoing treacherous, and powerful winds blow all vessels at sea off course. The temperature plummets, killing crops every week until the storm ceases.");
```
Comment (pawlick2) [inline backend/src/main/java/com/empire/World.java]: This also seems misleading about the grace period.  I may have actually missed some earlier, you might just want to check all of the descriptions if this comment is warranted.

File: backend/src/main/java/com/empire/World.java [hunk 5]
```diff
@@ -2065,13 +2088,10 @@ void evaluateGothiSpells() {
...
 				}
 			}
-			if (tivar.veil) {
-				for (Region r : regions) r.crops *= .97;
-			}
-			if (tivar.deluge) {
-				for (Region r : regions) r.crops *= .97;
-				
-			}
+			damageCrops(tivar.deluge);
```
Comment (pawlick2) [inline backend/src/main/java/com/empire/World.java]: I know this came up previously, I don't recall if this was the same context.  But damaging the crops this way doesn't additively stack the percentages when multiple Gothi spells or potential other crop-damaging effects happen in a region.  That in itself isn't a huge issue, but with the new progressive damage scale the order in which the spells' effects are applied also makes a difference depending on how longer-running spells are earlier/later in the order.

File: backend/src/main/java/com/empire/Constants.java [hunk 6]
```diff
@@ -72,6 +72,10 @@
 	public static final double riverOfKuunRecruitmentMod = 0.5;
 	public static final double riverOfKuunNeighborTaxMod = 0.5;
 
+	public static final double tivarSpellContinueChance = 0.66;
```
[the remark this answers is not in the document]
Comment (jpawlicki) [reply backend/src/main/java/com/empire/Constants.java]: The rules doc says 66%, not 2/3s, so for the moment being compliant with that.

File: backend/src/main/java/com/empire/World.java (hunk 5 above)
Comment (jpawlicki) [reply backend/src/main/java/com/empire/World.java]: That's correct, they're multiplicative. I can clarify that they are independent in the rules doc.

As for the order mattering, I don't think that's true, by the commutative property of multiplication, e.g. the net result here is 
crops = crops * (1-damage1) * (1-damage2) * (1-damage3) * (1-damage4)
and that should be the same regardless of what order they are applied in. Right?

File: backend/src/main/java/com/empire/World.java (hunk 4 above)
Comment (jpawlicki) [reply backend/src/main/java/com/empire/World.java]: Yep, I missed these four. Fixed.

File: backend/src/main/java/com/empire/World.java (hunk 3 above)
Comment (jpawlicki) [reply backend/src/main/java/com/empire/World.java]: Done.

File: backend/src/main/java/com/empire/World.java (hunk 2 above)
Comment (jpawlicki) [reply backend/src/main/java/com/empire/World.java]: Done.

File: frontend/orders-pane.js (hunk 1 above)
Comment (jpawlicki) [reply frontend/orders-pane.js]: Clarified.

File: backend/src/main/java/com/empire/World.java (hunk 5 above)
Comment (pawlick2) [reply backend/src/main/java/com/empire/World.java]: You're right, not sure what I was thinking at the time, I had just gotten back from running.

**Pull request merged**